### PR TITLE
Fix tracks color-by when properties change

### LINKED ISSUE #5330: tracks 'color_by' attribute not maintained after updating properties
## 🐛 Bug

The `color_by` attribute on a tracks layer changes if the properties are updated. 

## To Reproduce

```python
import napari
from skimage import data
import numpy as np

hubble_image = data.hubble_deep_field()

tracks_data = np.asarray([
    [1, 0, 236, 0],
    [1, 1, 236, 100],
    [1, 2, 236, 200],
    [1, 3, 236, 500],
    [1, 4, 236, 1000],
    [2, 0, 436, 0],
    [2, 1, 436, 100],
    [2, 2, 436, 200],
    [2, 3, 436, 500],
    [2, 4, 436, 1000],
    [3, 0, 636, 0],
    [3, 1, 636, 100],
    [3, 2, 636, 200],
    [3, 3, 636, 500],
    [3, 4, 636, 1000]
])
track_confidence = np.array(5*[0.9] + 5*[0.3] + 5 * [0.1])
properties = {
    'time': tracks_data[:, 1],
    'confidence': track_confidence
}

viewer = napari.view_image(hubble_image)
viewer.add_tracks(tracks_data, properties=properties)
napari.run()
```

then in the console

```ipython
viewer.layers['tracks_data'].color_by
Out[1]: 'track_id'

viewer.layers['tracks_data'].properties = viewer.layers['tracks_data'].properties

viewer.layers['tracks_data'].color_by
Out[3]: 'time'
```
## Expected behavior

If the previous 'color_by' attribute is still present in the properties, this should not be changed by updating the properties

## Additional context

The offending lines of code are 
https://github.com/napari/napari/blob/cd5c3147a13a42304a6e26a56f516bdc8b879557/napari/_qt/layer_controls/qt_tracks_controls.py#L128-L130

### PR DESCRIPTION
**Pull request by andy-sweet**
# Fixes/Closes
Closes #5330

# Description
This fixes two bugs related to `Tracks.color_by`.

The bug described in #5330 (the color-by property is still in the new properties dict) is fixed by blocking the signal associated when adding all the property keys as items to the color-by combobox.

I found a second related bug that occurs when the color-by property is no longer in the new properties dict. This is fixed by pushing the emission of `Tracks.events.properties` until after that problem has been fixed up.

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?
I added two integration tests for the Qt tracks controls and also manually tested the example in #5330.

### CODE REVIEW

**Comment by andy-sweet** [comment]
@alisterburt : assigning you to review as the original bug reporter.

@Kd-Here : we started this work several months ago. I was do some git-branch Spring cleaning and decided to push a fix up. Feel free to take a look and leave comments!

File: napari/_qt/layer_controls/_tests/test_qt_tracks_layer.py [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
 
 from napari._qt.layer_controls.qt_tracks_controls import QtTracksControls
 from napari.layers import Tracks
 
-_TRACKS = np.zeros((2, 4))
-_PROPERTIES = {'speed': [50, 30], 'time': [0, 1]}
 
+@pytest.fixture
+def data() -> np.ndarray:
+    return np.zeros((2, 4))
```
Comment (jni) [inline napari/_qt/layer_controls/_tests/test_qt_tracks_layer.py]: Can we perhaps rename this to `null_data` or `null_tracks_data`?

File: napari/_qt/layer_controls/_tests/test_qt_tracks_layer.py [hunk 2]
```diff
@@ -38,3 +49,43 @@ def test_tracks_controls_color_by(qtbot):
...
+def test_color_by_missing_after_properties_change(data, properties, qtbot):
+    """See https://github.com/napari/napari/issues/5330"""
+    layer = Tracks(data, properties=properties)
+    layer.color_by = 'time'
+    controls = QtTracksControls(layer)
+    qtbot.addWidget(controls)
+    assert controls.color_by_combobox.currentText() == 'time'
+
+    # Change the properties value by removing the time column.
+    with pytest.warns(UserWarning):
```
Comment (jni) [inline napari/_qt/layer_controls/_tests/test_qt_tracks_layer.py]: Add a match string here to make the test more readable?

**Review by jni — APPROVED** [review]
@andy-sweet I've made two very minor readability suggestions, but this is nice and clean, thanks! 😊

File: napari/_qt/layer_controls/_tests/test_qt_tracks_layer.py (hunk 1 above)
Comment (andy-sweet) [reply napari/_qt/layer_controls/_tests/test_qt_tracks_layer.py]: Renamed to `null_data`.

File: napari/_qt/layer_controls/_tests/test_qt_tracks_layer.py (hunk 2 above)
Comment (andy-sweet) [reply napari/_qt/layer_controls/_tests/test_qt_tracks_layer.py]: Done.

**Comment by andy-sweet** [comment]
Thanks for the review @jni! As this is relatively simple, I'll probably merge this after 24 hours unless I hear otherwise.

**Comment by psobolewskiPhD** [comment]
@andy-sweet If I'm reading right, this supercedes/closes https://github.com/napari/napari/pull/5343 right?

**Comment by andy-sweet** [comment]
> @andy-sweet If I'm reading right, this supercedes/closes #5343 right?

That's true. I closed that with a pointer back here.

**Comment by andy-sweet** [comment]
@alisterburt : removed you as a reviewer since @jni got there first. Feel free to leave a review if you're interested.

**Pull request merged**